include/mesh/mesh.h

```diff
@@ -82,7 +82,7 @@ class Mesh : public DefaultMesh
    * Copy and move assignment are not allowed.
    */
   Mesh & operator= (const Mesh &) = delete;
-  Mesh & operator= (Mesh &&) = delete;
+
```

Comment:
@jwpeterson, is getting rid of this enough?  I'd think we'd need to explicitly set it `= default` too.

Reply:
I guess it's enough for our immediate purposes - since we only use the move assignment via the virtual assign() function, and that's implemented at the ReplicatedMesh and DistributedMesh level so it explicitly calls those move assignment operators, and Mesh is a shim with no other data to move, this works.  And the new unit tests confirms that.

It might still be a good idea to add an = default here just as a courtesy to anyone who for some reason wants to do a move assignment of Mesh directly.  Or it might not, since that'd just be one more thing to add extra test coverage for...

Reply:
Yeah if I recall correctly, the compiler doesn't auto generate the special functions if you have customized any of them. I'd say even if that wasn't the case it would be better to explicitly document that are using default move assignment.